traced/ykrt/src/mt.rs

```diff
@@ -179,7 +186,7 @@ struct MTInner {
     hot_threshold: AtomicHotThreshold,
     active_user_threads: AtomicUsize,
     /// The ordered queue of compilation worker functions.
-    job_queue: Mutex<VecDeque<Box<dyn FnOnce() + Send>>>,
+    job_queue: (Condvar, Mutex<VecDeque<Box<dyn FnOnce() + Send>>>),
```

Comment:
The cond var is guarded by the mutex, so should we put it inside the mutex alongside the vecdeque?

They don't do this in the [rust docs](https://doc.rust-lang.org/beta/std/sync/struct.Condvar.html), so I expect to be wrong, but wouldn't this stop a thread from accessing the var without holding the lock?

Reply:
Note we're using `parking_lot` here https://docs.rs/parking_lot/0.11.1/parking_lot/struct.Condvar.html -- if we put the condvar in the mutex, I think that we wouldn't ever be able to unlock it to put a thread to sleep.